src/Couchbase.Lite.Shared/Query/XQuery.cs

```diff
@@ -58,7 +58,7 @@ internal class XQuery : IQuery
         private DateTime _lastUpdatedAt;
         private int _observingCount = 0;
         [NotNull]private Parameters _queryParameters = new Parameters();
-        private AtomicBool _willUpdate = false;
+        private AtomicBool _willUpdate = false, _updating = false, _stopping = false;
```

Comment:
FYI there is no reason to have these be atomic bools.  The reason to have an atomic bool is to be able to set and check the previous value as one atomic step (i.e. `if foo.set(value)`) so these two added ones could just be normal bools